controllers/helmchart_controller.go

```diff
@@ -490,9 +490,15 @@ func (r *HelmChartReconciler) reconcileFromTarballArtifact(ctx context.Context,
 		return sourcev1.HelmChartNotReady(chart, sourcev1.StorageOperationFailedReason, err.Error()), err
 	}
 
+	version := helmChart.Metadata.Version
+	// Uses GitRepository or Bucket revision if ignoring Chart.yaml version
+	if chart.Spec.IgnoreVersion {
+		version = artifact.Revision
+	}
```

Comment:
A similar block would be required in helm-controller (in addition to ensuring that the `Kind` is not `HelmRepository`).